### PR TITLE
fixed warnings [9193]

### PR DESCRIPTION
**Pull request by MiguelCompany**
Port of #1290 

* fixed warnings

Signed-off-by: ahcorde <ahcorde@gmail.com>

* Added cosmetic suggestions

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2508/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2321/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2515/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2509/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2322/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2516/)

**Review by IkerLuengo — DISMISSED** [review]
LGTM

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2601/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2417/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2611/)

File: include/fastrtps/utils/collections/ResourceLimitedVector.hpp [hunk 1]
```diff
@@ -461,7 +461,9 @@ class ResourceLimitedVector
      *
      * @return const reference to the underlying collection.
      */
-    operator const collection_type& () const noexcept { return collection_; }
+    operator const collection_type& () const noexcept {
+        return collection_;
+    }
```
Comment (IkerLuengo) [inline include/fastrtps/utils/collections/ResourceLimitedVector.hpp:466]:
```suggestion
    operator const collection_type& () const noexcept
    {
        return collection_;
    }
```

**Review by IkerLuengo — CHANGES_REQUESTED** [review]

**Review by IkerLuengo — DISMISSED** [review]
LGTM

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2910/)
* Linux aarch64Build Status
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2662/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2921/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3050/)
* Linux aarch64Build Status
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2733/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3061/)

**Review by IkerLuengo — APPROVED** [review]
LGTM

**Pull request merged**